docs/content/best-practices/github-workflow.md

```diff
@@ -0,0 +1,162 @@
+
+### Setting up secrets in your repository
+In order to publish your bundle to Docker Hub, you will need to set up secrets. You can add these in the settings of your repository and then use them in your workflow.
+
+### Using credential files
+If you were using credentials in your bundle, you will need to set up a credential file in your repository to use with your workflow. After you run `porter credentials generate`, you should see a JSON file with the name of your bundle in the ~/.porter/credentials directory. You need to add this file to your repository so you can pass in the file as the credential. For example, if my bundle was named hello, there would be a file with the credentials called hello.json. I would do the following to install my bundle:
+```yaml
+porter install -c ./hello.json
+```
+
+## Example Code
+
+Now, we will show example code for a workflow and then explain what the code does.
+
+* [Show example code](#show-example-code)
+* [Explain the code](#explain-the-code)
+
+### Show example code
+```yaml
+name: CI
+
+# Controls when the action will run. Triggers the workflow on push event for the master branch
+on:
+  push:
+    branches: [ main ]
+
+# Set up environment variables needed for the bundle
+env:
+  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+# A workflow run
+jobs:
+  publish: 
+    runs-on: ubuntu-latest
+    
+    steps: 
+    # Check out code
+    - uses: actions/checkout@v1
+    # Use Porter GH action to set up Porter
+    - name: Setup Porter
+      uses: deislabs/porter-gh-action@v0.1.1
+      with:
+        porter_version: v0.27.2
+    # Install docker mixin needed for the bundle
+    - name: Install Docker mixin
+      run: porter mixins install docker
+    # Run install
+    - name: Porter install
+      run: porter install -c ./docker-example.json --allow-docker-host-access
+    # Run upgrade
+    - name: Porter upgrade
+      run: porter upgrade -c ./docker-example.json --allow-docker-host-access
+    # Run uninstall
+    - name: Porter uninstall
+      run: porter uninstall -c ./docker-example.json --allow-docker-host-access
+    # Login to Docker Hub to publish the bundle
+    - uses: azure/docker-login@v1
+      name: Docker Login
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    # Run publish
+    - name: Porter Publish
+      run: porter publish
+```
+
+### Explain the code
+The first part of the code gives the workflow a name. Next, the on push explains that this workflow will run on a push to main. If you want it to run on a pull request, you can change this from push to pull_request. You can also change the branch name from main to the name of the branch you want the workflow to run on. 
```

Comment:
I think it would be easier if most of these explanations were inline comments in the example above. Some of the comments may not be necessary, like "Give the workflow a name" but others would be easier to understand if people could see it right next to the code in question rather than trying to follow along here and reference it back to the code above.